tests/test_iam_validation.py

```
@@ -0,0 +1,95 @@
+from sparkpilot.services.iam_validation import validate_assume_role_chain
+
+
+def test_validate_assume_role_chain_includes_external_id_from_settings(monkeypatch) -> None:
+    monkeypatch.setenv("SPARKPILOT_DRY_RUN_MODE", "false")
+    monkeypatch.setenv("SPARKPILOT_ASSUME_ROLE_EXTERNAL_ID", "tenant-external-id-123")
+    get_settings.cache_clear()
+
+    observed_kwargs: dict[str, object] = {}
+
+    class _FakeStsClient:
+        def assume_role(self, **kwargs):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `assume_role`

(ANN202)